src/components/CommentDeleter.jsx

```diff
@@ -0,0 +1,42 @@
+import { useState } from "react";
+import { deleteComment } from "../api";
+import ArticleComments from "./ArticleComments";
+
+export default function CommentDeleter({
+  comment,
+  articleComments,
+  setArticleComments,
+  isCommentDeleted,
+  setIsCommentDeleted,
+}) {
+  const [commentAuthor, setCommentAuthor] = useState("jessjelly");
+
+  const handleClick = (event) => {
+    let id = comment.comment_id;
+    event.currentTarget.disabled = true;
+
+    const updatedArticleComments = articleComments.filter(
+      (comment) => comment.comment_id !== id
+    );
+    setIsCommentDeleted(true)
+
+    setArticleComments(updatedArticleComments);
+    deleteComment(comment.comment_id);
+    
```

Comment:
What about if the delete fails? It could be quite important for the user to know that the comment is deleted.